### ISSUE TITLE
Allow authenticating auth routes with existing token instead of requiring a clientId and clientSecret

### ISSUE THREAD

**Issue by colin-oos** [boxcast/boxcast-sdk-js#58]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `@boxcast/boxcast-sdk-js@2.0.2` for the project I'm working on. I wanted to be able to use the sdk in node.js but I am using OAuth tokens rather than a clientId:clientSecret pair to authenticate. Currently your SDK only accepts a clientId and clientSecret to authenticate and generate a token, however since I already have a token I want to be able to just pass my existing token as the first argument. I expect this would be something others would need to be able to do as well so I decided to create this issue for further review. 

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/@boxcast/boxcast-sdk-js/src/api/auth_routes.ts b/node_modules/@boxcast/boxcast-sdk-js/src/api/auth_routes.ts
index 13600b7..1714ae3 100644
--- a/node_modules/@boxcast/boxcast-sdk-js/src/api/auth_routes.ts
+++ b/node_modules/@boxcast/boxcast-sdk-js/src/api/auth_routes.ts
@@ -21,7 +21,13 @@ export default class AuthenticatedRoutes {
     STATE.lastAuthToken = null;
   }
 
-  async authenticate(clientId, clientSecret) {
+  async authenticate(clientId, clientSecret?) {
+    if (clientId && !clientSecret) {
+      STATE.lastAuthToken = clientId;
+      return {
+        access_token: clientId,
+      }
+    }
     try {
       const body = new URLSearchParams({
         'grant_type': 'client_credentials',
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by hutch78** [comment]
@colin-oos  Hi Colin! My pleasure, it has been a fun project so far ^_^.

I like your suggestion of enabling the SDK to use a passed-in token instead of being forced to grab a new one just to store it in `STATE`. That said, rather than modifying the `authenticate()` method, I think we should keep `authenticate()` doing just that, and adding a `setToken` method that can be called when we already have a token. How does that sound?

If you like the idea, we can look at getting it into the next release of the SDK. In the mean time, feel free to submit a PR. 

Thanks for the feedback!

**Comment by colin-oos** [comment]
Yeah that makes sense. I'll submit a PR when I get a chance. Thanks!
